keras_spiking/layers.py

```diff
@@ -225,13 +223,13 @@ class SpikingActivationCell(KerasSpikingCell):
     .. testcode::
 
         my_layer = tf.keras.layers.RNN(
-            keras_spiking.SpikingActivationCell(units=10, activation=tf.nn.relu)
+            keras_spiking.SpikingActivationCell(input_size=10, activation=tf.nn.relu)
         )
 
     Parameters
     ----------
-    units : int
-        Dimensionality of layer.
+    input_size : tuple of int or ``tf.TensorShape``
```

Comment:
I'm wondering if we should just call it `size` for these user-facing classes. `input_size` implies that the `output_size` might be different, which is not the case.

Reply:
I think the worry is that `size` (to me) suggests the numpy sense of `size` (i.e. the flattened shape). For some reason I don't feel that as much with `input_size`, but I have no logical basis for that. Neither option seems great, maybe we should just do a quick poll and see what makes intuitive sense to people.

Reply:
Yeah, I definitely agree that it's annoying when "size" and "shape" aren't used consistently for flattened/unflattened descriptions of the dimensions. Even Numpy isn't perfect, though; for example, the `np.random` functions use `size` to specify the size of the output, but it accepts an int or a tuple.

Maybe we should change it to `shape`?

Reply:
Ok, I added a commit demoing what it looks like with "shape". The main downside is that it doesn't quite match `output_size` and `state_size`, which are special names so we can't change them. But I don't mind that.

Reply:
The problem with `shape` is that the somewhat loose Keras convention is that `shape` is the full shape, whereas `size` is the shape not including the batch/time dimensions. e.g. for an RNN, `output_shape` is `(batch_size, n_steps, d)`, while `output_size` is `(d,)`. Since in this case `input_size` doesn't (and shouldn't) include the batch/time dimensions, that's what made me want to use the `size` nomenclature.

Reply:
It *can* still be an integer (it doesn't have to be a tuple), and in many cases I think it will be. My inclination is to just call it `size`. I've made a poll in team-nengo.

Reply:
Ok, I've made the change to `size` as discussed in team-nengo. @drasmuss, let me know if all looks good, and I'll squash that into the "Support multidimensional" commit before I merge.